manifests/init.pp

```diff
@@ -211,12 +211,12 @@
   String $version = $foreman::params::version,
   Enum['installed', 'present', 'latest'] $plugin_version = $foreman::params::plugin_version,
   Boolean $db_manage = $foreman::params::db_manage,
-  Stdlib::Host $db_host = 'UNSET',
-  Variant[Undef, Enum['UNSET'], Stdlib::Port] $db_port = 'UNSET',
-  String $db_database = 'UNSET',
+  Optional[Stdlib::Host] $db_host = undef,
+  Optional[Stdlib::Port] $db_port = undef,
+  Optional[String] $db_database = undef,
   String $db_username = $foreman::params::db_username,
   String $db_password = $foreman::params::db_password,
```

Comment:
I tend to agree, but the code doesn't allow the password to be `undef`?

Reply:
(but also, postgresql::postgresql_password doesn't accept `undef` as a password)

Reply:
If you look at the old part of the diff, you'll see that it used to. Right now I aligned it to the data type here, but I wonder what we should be doing.

Reply:
Ah, that was a question… ;-)

I must admit I always found ident/peer based auth in PostgreSQL weird.

Like, today, we deploy the password to `/etc/foreman/database.yml`, but we don't set a hostname, so it uses a socket and thus peer auth, which succeeds as the process is running as the `foreman` user. I can set the password to something totally wrong and it will still work.

At the same time, I can not make `psql -U foreman -W` work as non-foreman, as it will ask for the password but then still use peer auth. Adding `-h localhost` helps, but oh well.

That said, I would prefer to get this is, as it fixes stuff, and then discuss alternative auth methods.

Reply:
It was indeed intended as opening a discussion for longer term, not as something blocking for this. I thought it was good to share my thoughts I had while writing this.